### ISSUE TITLE
NotImplementedException in TextSelection.VirtualSelectedSpans.get() while typing block comments

### ISSUE THREAD

**Issue by minestarks** [mono/monodevelop#4988]
- Build Monodevelop latest `master` commit f39391ed8f28461ace112906a74ba4bc269a51bc
- Build WebToolingAddin latest `master` commit 4ff1d20771f0775274ecf55aadffb918ad2e6345

Repro:
1. Open a new TypeScript file
2. Type `/*` and hit Enter

Exception (blocks typing):
```
System.NotImplementedException
  HResult=0x80004001
  Message=The method or operation is not implemented.
  Source=MonoDevelop.SourceEditor
  StackTrace:
   at Microsoft.VisualStudio.Text.Editor.Implementation.TextSelection.get_VirtualSelectedSpans() in E:\src\monodevelop\main\src\addins\MonoDevelop.SourceEditor2\VSEditor\TextSelection.cs:line 197
```

Callstack:
```
	MonoDevelop.SourceEditor.dll!Microsoft.VisualStudio.Text.Editor.Implementation.TextSelection.VirtualSelectedSpans.get() Line 197	C#
 	Microsoft.VisualStudio.Text.Implementation.dll!Microsoft.VisualStudio.Text.Operations.Implementation.EditorOperations.InsertText(string text, bool final, string undoText, bool isOverwriteModeEnabled) Line 3065	C#
 	Microsoft.VisualStudio.Text.Implementation.dll!Microsoft.VisualStudio.Text.Operations.Implementation.EditorOperations.InsertText(string text, bool final) Line 2875	C#
 	Microsoft.VisualStudio.Text.Implementation.dll!Microsoft.VisualStudio.Text.Operations.Implementation.EditorOperations.InsertText(string text) Line 1823	C#
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Implementation.BlockCommentEditing.BaseAbstractBlockCommentEditingCommandHandler.TryHandleReturnKey(Microsoft.VisualStudio.Text.ITextBuffer subjectBuffer, Microsoft.VisualStudio.Text.Editor.ITextView textView) Line 57	C#
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Implementation.BlockCommentEditing.AbstractBlockCommentEditingCommandHandler.ExecuteCommand(Microsoft.CodeAnalysis.Editor.Commands.ReturnKeyCommandArgs args, System.Action nextHandler) Line 36	C#
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Implementation.Commands.CommandHandlerService.ExecuteHandlers<Microsoft.CodeAnalysis.Editor.Commands.ReturnKeyCommandArgs>(System.Collections.Generic.IList<Microsoft.CodeAnalysis.Editor.ICommandHandler<Microsoft.CodeAnalysis.Editor.Commands.ReturnKeyCommandArgs>> commandHandlers, Microsoft.CodeAnalysis.Editor.Commands.ReturnKeyCommandArgs args, System.Action lastHandler) Line 102	C#
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Implementation.Commands.CommandHandlerService.Execute<Microsoft.CodeAnalysis.Editor.Commands.ReturnKeyCommandArgs>(Microsoft.VisualStudio.Utilities.IContentType contentType, Microsoft.CodeAnalysis.Editor.Commands.ReturnKeyCommandArgs args, System.Action lastHandler) Line 69	C#
 	MonoDevelop.CSharpBinding.dll!MonoDevelop.Ide.Completion.Presentation.RoslynCommandTarget.ExecuteReturn(System.Action executeNextCommandTarget) Line 104	C#
 	WebToolingAddin.dll!WebToolingAddin.JsTsEditorExtension.KeyPress(MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) Line 53	C#
 	MonoDevelop.Ide.dll!MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress(MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) Line 83	C#
 	MonoDevelop.Ide.dll!MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress(MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) Line 83	C#
 	MonoDevelop.Ide.dll!MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress(MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) Line 83	C#
 	MonoDevelop.Ide.dll!MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress(MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) Line 83	C#
 	MonoDevelop.SourceEditor.dll!MonoDevelop.SourceEditor.ExtensibleTextEditor.ExtensionKeyPress(Gdk.Key key, uint ch, Gdk.ModifierType state) Line 282	C#
 	MonoDevelop.SourceEditor.dll!MonoDevelop.SourceEditor.ExtensibleTextEditor.OnIMProcessedKeyPressEventInternal(Gdk.Key key, uint ch, Gdk.ModifierType state) Line 399	C#
 	MonoDevelop.SourceEditor.dll!MonoDevelop.SourceEditor.ExtensibleTextEditor.OnIMProcessedKeyPressEvent(Gdk.Key key, uint ch, Gdk.ModifierType state) Line 348	C#
 	MonoDevelop.SourceEditor.dll!Mono.TextEditor.TextArea.OnKeyPressEvent(Gdk.EventKey evt) Line 1289	C#
 	gtk-sharp.dll!Gtk.Widget.keypressevent_cb(System.IntPtr widget, System.IntPtr evnt)	Unknown
 	[Native to Managed Transition]	
 	[Managed to Native Transition]	
 	gtk-sharp.dll!Gtk.Widget.OnKeyPressEvent(Gdk.EventKey evnt)	Unknown
 	MonoDevelop.Ide.dll!MonoDevelop.Ide.Gui.DefaultWorkbench.OnKeyPressEvent(Gdk.EventKey evnt) Line 1218	C#
 	gtk-sharp.dll!Gtk.Widget.keypressevent_cb(System.IntPtr widget, System.IntPtr evnt)	Unknown
 	[Native to Managed Transition]	
 	[Managed to Native Transition]	
 	gtk-sharp.dll!Gtk.Application.Run()	Unknown
 	MonoDevelop.Ide.dll!MonoDevelop.Ide.IdeApp.Run() Line 426	C#
 	MonoDevelop.Ide.dll!MonoDevelop.Ide.IdeStartup.Run(MonoDevelop.Ide.MonoDevelopOptions options) Line 320	C#
 	MonoDevelop.Ide.dll!MonoDevelop.Ide.IdeStartup.Main(string[] args, MonoDevelop.Ide.Extensions.IdeCustomizer customizer) Line 693	C#
 	MonoDevelop.exe!MonoDevelop.Startup.MonoDevelopMain.Main(string[] args) Line 39	C#
```

> VS bug [#632385](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/632385)

**Comment by KirillOsenkov** [comment]
Ouch, how did this slip under the radar. We need to implement this. /cc @ToddGrun @DavidKarlas

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-23] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #5391 merged by: Kirill Osenkov  
 Author: mono  
 Commit: 55f70c9264f4093cbbb4373c2b830c8f0d0bfff3 (xamarin/monodevelop)
